common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java

```diff
@@ -128,6 +153,35 @@ public boolean isYoungerThanRetentionThreshold(long daysToRetain) {
     return this.rollingStartNumber >= threshold;
   }
 
+  /**
+   * Determines if this key is valid.
+   *
+   * @return true, if valid, else false.
+   */
+  public boolean isValid() {
+    Set<ConstraintViolation<DiagnosisKey>> violations = getConstraintViolations();
+    return violations.isEmpty();
+  }
+
+  /**
+   * Validates this key.
+   */
+  public void validate() {
+    Set<ConstraintViolation<DiagnosisKey>> violations = getConstraintViolations();
+
+    if (!violations.isEmpty()) {
```

Comment:
Shouldn't we reuse `isValid` here?

Reply:
I guess we want to log out the violations, hence we would need to call the `getConstraintViolations` again if we called `isValid` here.